### ISSUE TITLE
num_classes in create_models function

### ISSUE THREAD

**Issue by twinanda** [fizyr/keras-retinanet#678]
I am having a hard time understanding the code. So I just checked out today.
Now, I have a one class detection problem and I am creating my model using create_models function, passing num_classes = 1. However, while training, the generator gives two labels instead of one, and this gives an error in the training process. Some mismatched dimension at the loss layer.
I went through the code and found this in anchors.py:
`labels_batch      = np.zeros((batch_size, anchors.shape[0], num_classes + 1), dtype=keras.backend.floatx())`

Can anybody explain what the +1 is for here?

Seeing this, I changed my num_classes (manually) that I pass to create_models to 2. And as expected, it was training. I am not sure what my model is learning though because the model is now trained to detect two objects (?) even though only one object is in the dataset. 

Any help would be appreciated. Thank you.

**Comment by twinanda** [comment]
Issue resolved. The problem was in my code.
